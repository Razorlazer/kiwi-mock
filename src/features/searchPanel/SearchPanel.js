import * as React from 'react';
import AppBar from '@mui/material/AppBar';
import { Container, Grid, Divider, Button } from '@mui/material';
import Stack from '@mui/material/Stack';

import DepartureLocationSelector from './DepartureLocationSelector';
import DestinationLocationSelector from './DestinationLocationSelector';
import DateSelector from './DateSelector';
import FlightTypeSelector from './FlightTypeSelector';
import PriceSelector from './PriceSelector';
import useFlightsFetchHook from './useFlightsFetchHook';

const SearchPanel = () => {
    const { resetAndFetchFlights, isSearchDisabled} = useFlightsFetchHook();

    return (<AppBar position="static" color="inherit">
            <Container maxWidth="lg">
                    <Stack
                        mt={2}
                        mb={2}
                        spacing={1}
                        width={'100%'}
                        divider={<Divider orientation="horizontal" />}
                    >
                        <Grid container columnSpacing={2} rowSpacing={2} direction={'row'}>
                            <DepartureLocationSelector />
                            <DestinationLocationSelector />
                            <DateSelector />
                        </Grid>
                        <Grid container justifyContent="space-between" rowSpacing={1}>
                            <FlightTypeSelector/>
                            <PriceSelector/>
                            <Grid item lg={1} md={1} sm={12} xs={12}>
                                <Button color={'success'} fullWidth variant="outlined" onClick={resetAndFetchFlights} 
                                    disabled={isSearchDisabled} data-testid='search-button'>
                                    Search
                                </Button>
                            </Grid>
                        </Grid>
                    </Stack>
            </Container>
        </AppBar>)
}

export default SearchPanel;