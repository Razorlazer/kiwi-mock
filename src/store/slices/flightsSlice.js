import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import { fetchFlights } from '../../api';

export const flightsInitialState = {
    flightList: [],
    flightsSearchParams: { sort: 'price', limit: 10, offset: 0 },
    status: 'idle',
};

export const fetchFlightList = createAsyncThunk(
    'flights/list',
    async (params) => {
        const response = await fetchFlights(params);
        return response.data;
    }
);

export const flightsSlice = createSlice({
    name: 'flights',
    initialState: flightsInitialState,
    reducers: {
        changeFlightsSearchParams: (state, action) => {
            state.flightsSearchParams = {
                ...state.flightsSearchParams,
                ...action.payload
            };
        },
        changeFlightsSearchStatus: (state, action) => {
            state.status = action.payload;
        },
    },
    extraReducers: (builder) => {
        builder
            .addCase(fetchFlightList.pending, (state) => {
                state.status = 'loading';
            })
            .addCase(fetchFlightList.rejected, (state) => {
                state.status = 'rejected';
            })
            .addCase(fetchFlightList.fulfilled, (state, action) => {
                state.status = 'fullfilled';
                state.flightList = action.payload;
            });
    },
});

export const { changeFlightsSearchParams, changeFlightsSearchStatus } = flightsSlice.actions;
export const selectFlights = (state) => state.flights;
export default flightsSlice.reducer;
