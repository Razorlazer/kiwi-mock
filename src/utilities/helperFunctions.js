export const buildURLParams = (params = {})=>{
    const keys = Object.keys(params);

    const urlParams = [];
    for (let key = 0; key < keys.length; key++){
        if (params[keys[key]]){
            urlParams.push(`${keys[key]}=${params[keys[key]]}`)
        }
    }
  
    return urlParams.join('&');
};

export const createLocationSelectionList = ({ locations }, state) => {
    const mappedLocations = locations.filter(location => location.code !== state.destinationLocation && location.code!== state.departureLocation)
    .map(location => ({
        id: location.id,
        code: location.code,
        label: location.name,
        global_rank_dst: location.global_rank_dst,
        country: location.country,
        city: location.city
    }));

    return mappedLocations;
};

export const timeEpochUTCtoLocal = (dtime) => {
    var d = new Date(0); // The 0 there is the key, which sets the date to the epoch
    d.setUTCSeconds(dtime);
    return d.toLocaleString();
} 