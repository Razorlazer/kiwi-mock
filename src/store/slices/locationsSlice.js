import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import { fetchLocations } from '../../api';
import { createLocationSelectionList } from '../../utilities/helperFunctions';

const initialState = {
    departureLocationsList: [],
    destinationLocationsList: [],
    searchParams: { location_types: 'airport', term: 'Prague' },
    destinationLocation: null,
    departureLocation: null,
    departureLocationStatus: 'idle',
    destinationLocationStatus: 'idle',
};

export const fetchDepartureLocationsList = createAsyncThunk(
    'flights/departure/list',
    async (params) => {
        const response = await fetchLocations(params);
        return response.data;
    }
);

export const fetchDestinationLocationsList = createAsyncThunk(
    'flights/destination/list',
    async (params) => {
        const response = await fetchLocations(params);
        return response.data;
    }
);


export const locationsSlice = createSlice({
    name: 'flights',
    initialState,
    reducers: {
        changeLocationSearchParams: (state, action) => {
            state.searchParams = { ...state.searchParams, ...action.payload };
            state.status = 'loading';
        },
        changeDepartureLocation: (state, action) => {
            state.departureLocation = action.payload;
            state.status = 'loading';
        },
        changeDestinationaLocation: (state, action) => {
            state.destinationLocation = action.payload;
            state.status = 'loading';
        },
    },
    extraReducers: (builder) => {
        builder
            //Departure location actions
            .addCase(fetchDepartureLocationsList.pending, (state) => {
                state.departureLocationStatus = 'loading';
            })
            .addCase(fetchDepartureLocationsList.rejected, (state) => {
                state.departureLocationStatus = 'rejected';
            })
            .addCase(fetchDepartureLocationsList.fulfilled, (state, action) => {
                const list = createLocationSelectionList(action.payload, state);
                state.departureLocationStatus = 'fullfilled';
                state.departureLocationsList = list;
            })

            //Destination location actions
            .addCase(fetchDestinationLocationsList.pending, (state) => {
                state.destinationLocationStatus = 'loading';
            })
            .addCase(fetchDestinationLocationsList.rejected, (state) => {
                state.destinationLocationStatus = 'rejected';
            })
            .addCase(fetchDestinationLocationsList.fulfilled, (state, action) => {
                const list = createLocationSelectionList(action.payload, state);
                state.destinationLocationStatus = 'fullfilled';
                state.destinationLocationsList = list;
            });
    },
});

export const { changeLocationSearchParams, changeDepartureLocation, changeDestinationaLocation } = locationsSlice.actions;
export const selectLocations = (state) => state.locations;
export default locationsSlice.reducer;
