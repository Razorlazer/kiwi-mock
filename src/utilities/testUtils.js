import configureStore from 'redux-mock-store';
import { flightsInitialState } from '../store/slices/flightsSlice';
import { locationsInitialState } from '../store/slices/locationsSlice';

export const fullSearchState = {
    locations: {
        ...locationsInitialState,
        departureLocation: 'PRG',
        destinationLocation: 'VIE',
    },
    flights: flightsInitialState
}
export const mockStore = (initialState) =>{ 
    return configureStore([])(initialState || {
        locations: locationsInitialState,
        flights: flightsInitialState
    });
}

