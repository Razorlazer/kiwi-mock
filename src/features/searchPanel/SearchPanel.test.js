import { Provider } from 'react-redux';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import { mockStore, fullSearchState } from '../../utilities/testUtils';
import { fetchFlightList } from '../../store/slices/flightsSlice';
import SearchPanel from "./SearchPanel";


jest.mock('../../store/slices/flightsSlice', () => ({
    ...jest.requireActual('../../store/slices/flightsSlice'),
    fetchFlightList: jest.fn()
}));

afterEach(cleanup);


// jest.mock('./store/slices/locationsSlice', () => ({
//   ...jest.requireActual('./store/slices/locationsSlice'),
//   fetchDestinationLocationsList: jest.fn(),
//   fetchDepartureLocationsList: jest.fn()
// }));

describe('SearchPanel', () => {
    it('Should match the snapshot',()=>{
        const { asFragment } = render(
            <Provider store={mockStore()}>
                <SearchPanel />
            </Provider>
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('Should fetch flgihts on Search button click', () =>{
        const store = mockStore(fullSearchState);
        console.log(store.getState());
        render(
            <Provider store={store}>
                <SearchPanel />
            </Provider>
        );

        fireEvent.click(screen.getByTestId('search-button'));

        const dispatchedActions = store.getActions();
        console.log(dispatchedActions);
    })
});