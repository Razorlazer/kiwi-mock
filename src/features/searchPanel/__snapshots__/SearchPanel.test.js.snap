// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel Should match the snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorInherit MuiAppBar-positionStatic css-1cjrmu8-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="css-cdm1oc-MuiStack-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-o1xwpy-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 css-sktcqb-MuiGrid-root"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="location-input"
                  id="location-input-label"
                >
                  From
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-invalid="false"
                    autocapitalize="none"
                    autocomplete="new-password"
                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="location-input"
                    role="combobox"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Open"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                      tabindex="-1"
                      title="Open"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        From
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 css-sktcqb-MuiGrid-root"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="location-input"
                  id="location-input-label"
                >
                  To
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-invalid="false"
                    autocapitalize="none"
                    autocomplete="new-password"
                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="location-input"
                    role="combobox"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Open"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                      tabindex="-1"
                      title="Open"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        To
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 css-sktcqb-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r4:"
                id=":r4:-label"
              >
                Departure
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  aria-label="Choose date"
                  aria-readonly="true"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r4:"
                  readonly=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Departure
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 css-sktcqb-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r6:"
                id=":r6:-label"
              >
                Return
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  aria-label="Choose date"
                  aria-readonly="true"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r6:"
                  readonly=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Return
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-yk10cl-MuiDivider-root"
        />
        <div
          class="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-14xmr3p"
            >
              <div
                class="MuiToggleButtonGroup-root css-1gjgmky-MuiToggleButtonGroup-root"
                role="group"
              >
                <button
                  aria-pressed="true"
                  class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeSmall MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
                  tabindex="0"
                  type="button"
                  value="M"
                >
                  Economy
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-pressed="false"
                  class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
                  tabindex="0"
                  type="button"
                  value="W"
                >
                  Premium Economy
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-pressed="false"
                  class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
                  tabindex="0"
                  type="button"
                  value="C"
                >
                  Business
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-pressed="false"
                  class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1dlyelz-MuiButtonBase-root-MuiToggleButton-root"
                  tabindex="0"
                  type="button"
                  value="F"
                >
                  First class
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
                  id="slider"
                >
                  Price up to:
                </label>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-1v5z18m"
                >
                  <span
                    class="MuiSlider-colorPrimary MuiSlider-sizeSmall MuiSlider-root MuiSlider-marked css-xmer36-MuiSlider-root"
                  >
                    <span
                      class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
                    />
                    <span
                      class="MuiSlider-track css-1n40zqk-MuiSlider-track"
                      style="left: 0%; width: 25%;"
                    />
                    <span
                      class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
                      data-index="0"
                      style="left: 5%;"
                    />
                    <span
                      aria-hidden="true"
                      class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
                      data-index="0"
                      style="left: 5%;"
                    >
                      100€
                    </span>
                    <span
                      class="MuiSlider-mark MuiSlider-markActive css-6cwnna-MuiSlider-mark"
                      data-index="1"
                      style="left: 25%;"
                    />
                    <span
                      aria-hidden="true"
                      class="MuiSlider-markLabel MuiSlider-markLabelActive css-1eoe787-MuiSlider-markLabel"
                      data-index="1"
                      style="left: 25%;"
                    >
                      500€
                    </span>
                    <span
                      class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
                      data-index="2"
                      style="left: 50%;"
                    />
                    <span
                      aria-hidden="true"
                      class="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
                      data-index="2"
                      style="left: 50%;"
                    >
                      1000€
                    </span>
                    <span
                      class="MuiSlider-mark css-1tfve6q-MuiSlider-mark"
                      data-index="3"
                      style="left: 100%;"
                    />
                    <span
                      aria-hidden="true"
                      class="MuiSlider-markLabel css-yafthl-MuiSlider-markLabel"
                      data-index="3"
                      style="left: 100%;"
                    >
                      2000€
                    </span>
                    <span
                      class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeSmall MuiSlider-thumb css-14gf62f-MuiSlider-thumb"
                      data-focusvisible="false"
                      data-index="0"
                      style="left: 25%;"
                    >
                      <input
                        aria-label="slider"
                        aria-orientation="horizontal"
                        aria-valuemax="2000"
                        aria-valuemin="0"
                        aria-valuenow="500"
                        aria-valuetext="500€"
                        data-index="0"
                        max="2000"
                        min="0"
                        step="10"
                        style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                        type="range"
                        value="500"
                      />
                      <span
                        aria-hidden="true"
                        class="MuiSlider-valueLabel css-1xff9fg-MuiSlider-valueLabel"
                      >
                        <span
                          class="MuiSlider-valueLabelCircle"
                        >
                          <span
                            class="MuiSlider-valueLabelLabel"
                          >
                            500
                          </span>
                        </span>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root css-vj1n65-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-sxix9q-MuiButtonBase-root-MuiButton-root"
              data-testid="search-button"
              disabled=""
              tabindex="-1"
              type="button"
            >
              Search
            </button>
          </div>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
